api/controllers/v1/jobs.ts

```diff
@@ -134,11 +140,36 @@ async function stopECSTask(taskId: string, consoleLogger: ConsoleLogger) {
   await ecs.stopZombieECSTask(taskId);
 }
 
-async function retry(message: JobQueueMessage, consoleLogger: ConsoleLogger, url: string): Promise<any> {
+/**
+ * Type guard to check whether or not the val is a number.
+ * This performs the necessary validation to ensure that a given value is
+ * a number at runtime
+ * @param val the value we want to validate
+ * @returns a boolean that states whether or not the val is a number
+ */
+function isNumber(val: unknown): val is number {
+  if (typeof val !== 'number') {
+    if (typeof val === 'string') {
+      return !isNaN(parseInt(val));
+    }
+    return false;
+  }
+
+  return true;
+}
+
+async function retry(message: JobQueueMessage, consoleLogger: ConsoleLogger, url: string): Promise<void> {
```

Comment:
This isn't used, but I am not sure if there is value in keeping it around still. Any thoughts on keeping this function around?

Reply:
I think there is little value in keeping it - we used to try all failed builds three times (I don't recall why) but it was very annoying and provided no value. I'm pro removing unused code.